packages/web3-types/src/user_operation_types.ts

```diff
@@ -0,0 +1,36 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+export interface UserOperation {
+	sender: string;
+	nonce: string;
+	initCode: string;
+	callData: string;
+	callGasLimit: string;
+	verificationGasLimit: string;
+	preVerificationGas: string;
+	maxFeePerGas: string;
+	maxPriorityFeePerGas: string;
+	paymasterAndData: string;
+	signature: string;
```

Comment:
```suggestion
	sender: Address;
	nonce: Uint256;
	initCode: HexStringBytes;
	callData: HexStringBytes;
	callGasLimit?: Uint256;
	verificationGasLimit: Uint256;
	preVerificationGas: Uint256;
	maxFeePerGas?: Uint256;
	maxPriorityFeePerGas?: Uint256;
	paymasterAndData: HexStringBytes;
	signature: HexStringBytes;
```
you can import these types from src/eth_types.ts

also gas params are optional for user as mentioned in EIP

Reply:
Thx, fixed on [commit](https://github.com/web3/web3.js/pull/6302/commits/870e8c1cb2cc3603fe8b59ec87a4ca8106eafbe1#diff-50b6e0d939da1d7a07622f38ed21acf0f6c993f3b4b27aace5c8c7dbfaadc011)